File: 724-FindPivotIndex/724-FindPivotIndex.py
# Last updated: 4/29/2026, 10:28:28 AM
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums);
        sums = []
        sum = 0
        for i in range(n):
            sum += nums[i]
            sums.append(sum)
        print(sums)
        
        for i in range(n):
            leftIdx = i - 1
            rightIdx = i + 1
            leftSum = 0
            rightSum = 0
            if leftIdx >= 0:
                leftSum = sums[leftIdx]
            if rightIdx <= n - 1:
                rightSum = sums[n-1] - sums[i]
            if leftSum == rightSum:
                return i
        return -1
            

File: 724-FindPivotIndex/724-FindPivotIndex.js
// Last updated: 4/29/2026, 10:28:25 AM
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let sum = 0;
    for(const num of nums) {
        sum += num;
    }
    console.log(`sum: ${sum}`)
    let left = 0;
    let right = sum;
    for(let i = 0; i < nums.length; i++) {
        console.log(`i: ${i}, left=${left}, right=${right}`)
        left += i-1 >= 0? nums[i-1]: 0;
        right -= nums[i];
        if(left === right) {
            return i;
        }
    }
    return -1;
};